frontend/src/components/widgets/DataFrame/columns/DateTimeColumn.ts

```diff
@@ -0,0 +1,374 @@
+    // Default parameters:
+    {
+      format: defaultFormat,
+      step: defaultStep,
+      timezone,
+    },
+    // User parameters:
+    props.columnTypeOptions
+  ) as DateTimeColumnParams
+
+  let defaultTimezoneOffset: number | undefined = undefined
+  if (notNullOrUndefined(parameters.timezone)) {
+    // We try to determine the timezone offset based on today's date
+    // This is needed for the date picker to work correctly when the value is null
+    // TODO(lukasmasuch): But this might not be correct for dates in the past or future
+    // since the timezone offset might have changed based on a timezone name.
+    try {
+      defaultTimezoneOffset =
+        applyTimezone(moment(), parameters.timezone)?.utcOffset() || undefined
+    } catch (error) {
+      // Do nothing
+    }
+  }
+
+  let minDate: Date | undefined = undefined
+  if (notNullOrUndefined(parameters.min_value)) {
+    minDate = toSafeDate(parameters.min_value) || undefined
+  }
+
+  let maxDate: Date | undefined = undefined
+  if (notNullOrUndefined(parameters.max_value)) {
+    maxDate = toSafeDate(parameters.max_value) || undefined
+  }
+
+  const cellTemplate = {
+    kind: GridCellKind.Custom,
+    allowOverlay: true,
+    copyData: "",
+    readonly: !props.isEditable,
+    contentAlign: props.contentAlignment,
+    style: props.isIndex ? "faded" : "normal",
+    data: {
+      kind: "date-time-cell",
+      date: undefined,
+      displayDate: "",
+      step: parameters.step?.toString() || "1",
+      format: inputType,
+      min: minDate,
+      max: maxDate,
+    },
+  } as DateTimeCell
+
+  const validateInput = (data?: any): boolean | Date => {
+    const cellData: Date | null | undefined = toSafeDate(data)
+    if (cellData === null) {
+      if (props.isRequired) {
+        return false
+      }
+      return true
+    }
+
+    if (cellData === undefined) {
+      // Input cannot be interpreted as a date
+      return false
+    }
+
+    // Apply min_value configuration option:
+    if (
+      notNullOrUndefined(minDate) &&
+      // We compare on a string level so that it also works correctly for time and date values
+      toISOString(cellData) < toISOString(minDate)
+    ) {
+      return false
+    }
+
+    // Apply min_value configuration option:
+    if (
+      notNullOrUndefined(maxDate) &&
+      toISOString(cellData) > toISOString(maxDate)
+    ) {
+      return false
+    }
+
+    // TODO: validate step size
+
+    return true
+  }
+
+  return {
+    ...props,
+    kind,
+    sortMode: "default",
+    validateInput,
+    getCell(data?: any, validate?: boolean): GridCell {
+      if (validate === true) {
+        const validationResult = validateInput(data)
+        if (validationResult === false) {
+          // The input is invalid, we return an error cell which will
+          // prevent this cell to be inserted into the table.
+          return getErrorCell(toSafeString(data), "Invalid input.")
```

Comment:
Should we add anymore information to this error cell?

Reply:
Hmm, this is mostly an internal error that prevents the edit/pasted value to be submitted. We don't really have more specific information about why it is invalid at that point, but we could add some more information to report this issue.